Content.Server/Fluids/EntitySystems/FluidSpreaderSystem.cs

```diff
@@ -100,49 +101,57 @@ public override void Update(float frameTime)
     /// </summary>
     /// <param name="srcUid">Entity Uid of original puddle</param>
     /// <param name="srcPuddle">PuddleComponent attached to srcUid</param>
-    /// <param name="pos">at which to check tile</param>
+    /// <param name="dstPos">at which to check tile</param>
     /// <param name="mapGrid">helper param needed to extract entities</param>
-    /// <param name="puddle">either found or newly created PuddleComponent.</param>
+    /// <param name="newPuddleUid">either found or newly created PuddleComponent.</param>
     /// <returns>true if tile is empty or occupied by a non-overflowing puddle (or a puddle close to being overflowing)</returns>
-    private bool CheckTile(EntityUid srcUid, PuddleComponent srcPuddle, EntityCoordinates pos, MapGridComponent mapGrid,
-        [NotNullWhen(true)] out PuddleComponent? puddle)
+    private bool CheckTile(EntityUid srcUid, PuddleComponent srcPuddle, EntityCoordinates dstPos, 
+        MapGridComponent mapGrid, EntityQuery<PuddleComponent> puddleQuery,
+        [NotNullWhen(true)] out EntityUid? newPuddleUid, [NotNullWhen(true)] out PuddleComponent? newPuddleComp)
     {
-        if (!mapGrid.TryGetTileRef(pos, out var tileRef)
+        if (!mapGrid.TryGetTileRef(dstPos, out var tileRef)
             || tileRef.Tile.IsEmpty)
         {
-            puddle = null;
+            newPuddleUid = null;
+            newPuddleComp = null;
             return false;
         }
 
-        var puddleCurrentVolume = _puddleSystem.CurrentVolume(srcUid, srcPuddle);
+        // check if puddle can spread there at all
+        var dstMap = dstPos.ToMap(EntityManager, _transform);
+        var dst = dstMap.Position;
+        var src = Transform(srcUid).MapPosition.Position;
+        var dir = src - dst;
+        var ray = new CollisionRay(dst, dir.Normalized, (int) CollisionGroup.TabletopMachineMask);
```

Comment:
Why tabletop mask? Shouldn't it be something like `LowImpassable | Impassable`?

Reply:
@metalgearsloth wanted to change it

Reply:
Try not to use the composite masks unless there's a good reason, just go with a flag of individual layers.